pkg/network/setup/configstatecache_test.go

```diff
@@ -0,0 +1,126 @@
+/*
+* This file is part of the KubeVirt project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* Copyright 2023 Red Hat, Inc.
+*
+ */
+
+package network_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	network "kubevirt.io/kubevirt/pkg/network/setup"
+
+	dutils "kubevirt.io/kubevirt/pkg/ephemeral-disk-utils"
+
+	"kubevirt.io/kubevirt/pkg/network/cache"
+)
+
+var _ = Describe("config state cache", func() {
+	const (
+		uid     = "123"
+		testNet = "testnet"
+	)
+	var (
+		configStateCache network.ConfigStateCache
+		baseCacheCreator tempCacheCreator
+		ifacesMap        map[string]cache.PodIfaceState
+	)
+
+	BeforeEach(func() {
+		dutils.MockDefaultOwnershipManager()
+		ifacesMap = map[string]cache.PodIfaceState{}
```

Comment:
It would have been nicer if we could test everything without this additional data helper.

We have control on the persisted cache, so we can check if data is retrieved from it or not easily (by either adding specific data there or deleting it).

I guess this can be done as a follow up as well.